web/client-api/src/main/java/io/deephaven/web/client/api/filter/FilterValue.java

```diff
@@ -45,20 +50,54 @@ public static FilterValue ofString(@TsTypeRef(Any.class) Object input) {
         return new FilterValue(lit);
     }
 
+    @TsUnion
+    @JsType(name = "?", namespace = JsPackage.GLOBAL, isNative = true)
+    public interface OfNumberUnionParam {
+        @JsOverlay
+        static OfNumberUnionParam of(@DoNotAutobox Object value) {
+            return Js.cast(value);
+        }
+
+        @JsOverlay
+        default boolean isLongWrapper() {
+            return this instanceof LongWrapper;
+        }
+
+        @JsOverlay
+        default boolean isNumber() {
+            return (Object) this instanceof Double;
+        }
+
+        @TsUnionMember
+        @JsOverlay
+        default LongWrapper asLongWrapper() {
+            return Js.cast(this);
+        }
+
+        @TsUnionMember
+        @JsOverlay
+        default double asNumber() {
+            return Js.asDouble(this);
+        }
+    }
+
     @JsMethod(namespace = "dh.FilterValue")
-    public static FilterValue ofNumber(@TsTypeRef(Any.class) Object input) {
+    public static FilterValue ofNumber(OfNumberUnionParam input) {
```

Comment:
Note - the current API seems to also accept `BigDecimalWrapper` and `BigIntegerWrapper`, though there seems to be something incorrect with these filters at the moment (at least from the context menu and quick filters) on the latest main. I've filed an issue for it: https://github.com/deephaven/web-client-ui/issues/1216
I think the `OfNumberUnionParam` should also support BigDecimalWrapper/BigIntegerWrapper, since they get converted to strings and handled that way?